keras_nlp/models/task.py

```diff
@@ -267,59 +274,162 @@ def from_preset(
                 "constructor with a `backbone` argument. "
                 f"Received: backbone={kwargs['backbone']}."
             )
+
+        # Load backbone from preset.
+        config_path = os.path.join(preset, CONFIG_FILE)
```

Comment:
I think we need to rework this to a slightly new flow. Something like this...

```python
# Backbone case.
if not exists("task.json") or not issubclass(check_config_class("task.json"), cls):
    # This should be basically what is here already.
    # Load a preprocessor. Load a backbone.
    return cls(backbone=backbone, preprocessor=preprocessor, **kwargs)

# Task case.
task = keras.saving.deserialize_keras_object("task.json")
if load_weights:
    task.backbone.load_weights("model.weights.h5")
    task.load_task_weights("task.weights.h5")
```

Basically if we don't see a `task.json` or the `task.json` is for a different task, we load the low level objects and make a default task object with them. If we find a task object for our class, we load it exactly as it was before.

Let me know if that makes sense. I think the logic here is a bit different.

Reply:
What I was thinking was to load the backbone and preprocessor in any case (whether we have a `task.json` or not).
* If there is no `task.json` make a default task object with backbone and preprocessor. 
* If there is a `task.json`, load the task-specific things and assign the loaded `backbone` to `task.backbone`. 
Do you see any issues or disadvantages with this approach?
I'm mainly doing this because I thought it would be cleaner to just have one piece of code that loads the backbone.

PS: I made some changes this morning so they may not have been included in the version of code that you reviewed.

Reply:
Assigning the backbone to task would mean we double up on backbone memory until the next GC. We have to avoid that I think, otherwise we will OOM people very easily.

One case I was getting at in my snippet but worth calling out explicitly. You saved, say, a BertClassifier but are loading a different task, e.g. a BertMaskedLM. In this case our `task.json` is from the wrong object, and I think we fall back to the "backbone case" here.

Reply:
Oh, I see! Makes sense! Thanks for explaining this, Matt!